src/lib/storage/deprecated_dictionary_column.hpp

```diff
@@ -31,10 +32,10 @@ class DeprecatedDictionaryColumn : public BaseDeprecatedDictionaryColumn {
    * Creates a Dictionary column from a given dictionary and attribute vector.
    * See deprecated_dictionary_compression.cpp for more.
    */
-  explicit DeprecatedDictionaryColumn(pmr_vector<T>&& dictionary,
```

Comment:
Okay so your code here implies that `DictionaryColumn` will also be instantiated for `FixedString`, which means we need to be aware of it when resolving the type of a `BaseColumn`.  Most operators use functions in `resolve_type.hpp` to do this. These functions use a data structure defined in `all_type_variant.hpp` that maps values of `DataType` to actual types. It seems you don’t extend this data structure therefore `DictionaryColumn<FixedString>` will never be resolved therefore your code is broken.